module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -202,18 +202,22 @@ public function homeAction()
             try {
                 if (!$this->getAuthManager()->getIdentity()) {
                     $this->getAuthManager()->login($this->getRequest());
+
                     // Return early to avoid unnecessary processing if we are being
                     // called from login lightbox and don't have a followup action or
                     // followup is set to referrer.
                     if (
-                        $this->params()->fromPost('processLogin')
-                        && $this->inLightbox()
-                        && (!$this->hasFollowupUrl()
-                        || $this->followup()->retrieve('isReferrer') === true)
+                        $this->inLightbox()
```

Comment:
I'm not totally sure I understand why all of these changes were made. Are these all intentional or is there a possibility of merge errors here?

Reply:
This is a simplification of the logic. `$this->params()->fromPost('processLogin')` is redundant from line 197 above and I changed the whitespace to make the remaining logic more clear. This should be equivalent, it's a strictly stylistic (and unnecessary) change.

Reply:
In line 197, `$this->params()->fromPost('processLogin')` is part of a multi-part condition with several different OR'ed clauses. Thus, I don't think it's necessarily redundant here, because the existing code will skip this block if any of the other conditions (e.g. single-sign on session initiator, or explicit passing of an auth method) are met. Maybe that's what we need or want, but it's definitely not an entirely equivalent refactoring. I'd have to refresh my memory on the purpose of all those other code paths, and test everything, to be sure. If this was done for purely cosmetic reasons, I'd recommend reverting it just to save us work. But if we need it, then we should be sure we test every possibility.

Reply:
I've set this back to the original.